src/RestServerGateway.php

```diff
@@ -0,0 +1,98 @@
+<?php
+
+namespace Omnipay\SagePay;
+
+use Omnipay\SagePay\Message\ServerRestCompletePurchaseRequest;
+use Omnipay\SagePay\Message\ServerRestMerchantSessionKeyRequest;
+use Omnipay\SagePay\Message\ServerRestPurchaseRequest;
+use Omnipay\SagePay\Message\ServerRestRefundRequest;
+use Omnipay\SagePay\Message\ServerRestRepeatRequest;
+use Omnipay\SagePay\Message\ServerRestRetrieveTransactionRequest;
+use Omnipay\SagePay\Message\ServerRestVoidRequest;
+
+/**
+ * Sage Pay Rest Server Gateway
+ */
+class RestServerGateway extends ServerGateway
+{
+    public function getName()
+    {
+        return 'Sage Pay REST Server';
+    }
+    
+    public function getUsername()
+    {
+        return $this->getParameter('username');
+    }
+
+    public function setUsername($value)
+    {
+        return $this->setParameter('username', $value);
+    }
+
+    public function getPassword()
+    {
+        return $this->getParameter('password');
+    }
+
+    public function setPassword($value)
+    {
+        return $this->setParameter('password', $value);
+    }
+
+    /**
+     * Create merchant session key (MSK).
+     */
+    public function createMerchantSessionKey(array $parameters = array())
+    {
+        return $this->createRequest(ServerRestMerchantSessionKeyRequest::class, $parameters);
+    }
+
```

Comment:
Do the `authorize` method inherited from `ServerGateway` and the `capture` and `abort` methods inherited from `DirectGateway` work for the PI integration, or should these be overridden here and new `ServerRestAuthorizeRequest` etc be created?

Reply:
I just use purchase() and complete() from RestServerGateway.php at the moment. Authorize, capture and abort would need adding for the PI/REST integration